instrumentation/opentelemetry-instrumentation-redis/src/opentelemetry/instrumentation/redis/__init__.py

```diff
@@ -281,7 +288,13 @@ async def _async_traced_execute_pipeline(func, instance, args, kwargs):
                 span.set_attribute(
                     "db.redis.pipeline_length", len(command_stack)
                 )
-            response = await func(*args, **kwargs)
+
+            response = None
+            try:
+                response = func(*args, **kwargs)
```

Comment:
you lost an await here, do we have tests for async?

Reply:
you are right! I should add await here. Thanks!

I have added the async test, but seems without await here, the test will still pass.

this remind me that I should add sync test of WatchError as well, will update it in newest commit.